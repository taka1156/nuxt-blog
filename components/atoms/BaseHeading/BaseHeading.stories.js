import BaseHeading from './BaseHeading.vue';

export default {
  title: 'atoms/BaseHeading',
  component: BaseHeading,
  argTypes: {
    hLv: {
      control: {
        type: 'inline-radio',
        options: ['1', '2', '3', '4', '5', '6']
      }
    }
  }
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { BaseHeading },
  template: '<base-heading v-bind="$props">ダミーテキスト</base-heading>'
});

export const Default = Template.bind({});
Default.args = { hLv: '1' };
