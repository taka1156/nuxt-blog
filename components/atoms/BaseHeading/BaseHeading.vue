<template>
  <div>
    <component
      :is="`h${hLv}`"
      class="base-heading"
      :class="`base-heading${hLv}--extend`"
    >
      <!-- @slot headingタグの内容 -->
      <slot />
    </component>
  </div>
</template>

<script>
export default {
  name: 'BaseHeading',
  props: {
    /**
     * headingレベル(1~6)
     */
    hLv: {
      type: String,
      required: true,
      validator: function (value) {
        return [1, 2, 3, 4, 5, 6].indexOf(Number(value)) !== -1;
      }
    }
  }
};
</script>

<style scoped>
.base-heading {
  word-break: break-word;
  overflow-wrap: break-word;
}
</style>
