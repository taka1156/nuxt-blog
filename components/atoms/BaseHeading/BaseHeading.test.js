import { mount } from '@vue/test-utils';
import BaseHeading from './BaseHeading.vue';

describe('BaseImg', () => {
  const baseHeading = propsData =>
    mount(BaseHeading, {
      propsData: { ...propsData },
      slots: {
        default: 'ダミーの見出し'
      }
    });

  it('値がDOMに反映されているか(h1)', () => {
    const wrapper = baseHeading({ hLv: '1' });
    const hTag = wrapper.find('h1');
    expect(hTag.text()).toBe('ダミーの見出し');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('値がDOMに反映されているか(h3)', () => {
    const wrapper = baseHeading({ hLv: '3' });
    const hTag = wrapper.find('h3');
    expect(hTag.text()).toBe('ダミーの見出し');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});
