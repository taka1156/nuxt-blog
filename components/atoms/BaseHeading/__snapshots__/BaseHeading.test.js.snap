// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseImg 値がDOMに反映されているか(h1) 1`] = `
"<div>
  <h1 class=\\"base-heading base-heading1--extend\\">ダミーの見出し</h1>
</div>"
`;

exports[`BaseImg 値がDOMに反映されているか(h3) 1`] = `
"<div>
  <h3 class=\\"base-heading base-heading3--extend\\">ダミーの見出し</h3>
</div>"
`;
